pyne/partisn.py

```diff
@@ -0,0 +1,816 @@
+    if "/" in hdf5:
+        title = hdf5.split("/")[len(hdf5.split("/"))-1].split(".")[0]
+    else:
+        title = hdf5.split(".")[0]
+    
+    # call function to write to file
+    if input_file_tf:
+        _write_input(title, block01, block02, block03, block04, block05, name=input_file)
+    else:
+        _write_input(title, block01, block02, block03, block04, block05)
+  
+
+def _get_material_lib(hdf5, data_hdf5path, nuc_hdf5path, **kwargs):
+    """Read material properties from the loaded dagmc geometry.
+    """
+    
+    # If a set of nuc_names is provided, then collapse elements
+    if 'nuc_names' in kwargs:
+        nuc_names = kwargs['nuc_names']
+        collapse = True
+        # set of exception nuclides for collapse_elements
+        mat_except = Set(nuc_names.keys())
+    else:
+        collapse = False
+    
+    # collapse isotopes into elements (if required)
+    mats = MaterialLibrary(hdf5, datapath=data_hdf5path, nucpath=nuc_hdf5path)
+    mats_collapsed = {}
+    for mat_name in mats.keys():
+        if collapse:
+            mats_collapsed[mat_name] = mats[mat_name].collapse_elements(mat_except)
+        else:
+            mats_collapsed[mat_name] = mats[mat_name]
+
+    # convert mass fraction to atom density in units [at/b-cm]
+    mat_lib = {}
+    for mat_name, comp in mats_collapsed.iteritems():
+        atom_dens_dict = comp.to_atom_dens()
+        comp_list = {}
+        for nucid, dens in atom_dens_dict.iteritems():
+            # convert from [at/cc] to [at/b-cm]
+            comp_list[nucid] = dens*10.**-24
+        mat_lib[mat_name] = comp_list
+
+    return mat_lib
+
+
+def _nucid_to_xs(mat_lib, **kwargs):
+    """Replace nucids with xs library names.
+    """
+    
+    if 'nuc_names' in kwargs:
+        nuc_names = kwargs['nuc_names']
+        names_tf = True
+    else:
+        names_tf = False
+    
+    mat_xs_names = {}
+    for mat in mat_lib.keys():
+        mat_name = strip_mat_name(mat)
+        mat_xs_names[mat_name] = {}
+        for nucid in mat_lib[mat].keys():
+            if names_tf:
+                if nucid in nuc_names.keys():
+                    name = nuc_names[nucid]
+                    mat_xs_names[mat_name][name] = mat_lib[mat][nucid]
+                else:
+                    warn("Nucid {0} does not exist in the provided nuc_names dictionary.".format(nucid))
+                    mat_xs_names[mat_name]["{0}".format(nucid)] = mat_lib[mat][nucid]
+            else:
+                mat_xs_names[mat_name][nucname.name(nucid)] = mat_lib[mat][nucid]
+
+    return mat_xs_names
+    
+
+def _get_xs_names(mat_xs_names):
+    """Create list of names (strings) of the nuclides that appear in the cross
+    section library from the list of nuc_names.
+    """
+    
+    xs_names = []
+    for mat, nuc_set in mat_xs_names.iteritems():
+        for name in nuc_set.keys():
+            if name not in xs_names:
+                xs_names.append(name)
+    
+    return xs_names
```

Comment:
This is basically a set operation so I would write the body of this function as:

```python
xs_names = set()
list(map(xs_names.update, mat_xs_name.values()))
return list(xs_names)
```